core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -293,20 +294,19 @@ public static boolean getXOptAppCorrelationId(final Message msg) {
      */
     public static JsonObject getJsonPayload(final Message msg) {
 
-        final String payload = getPayload(msg);
-        return (payload != null ? new JsonObject(payload) : null);
+        final Buffer buffer = getPayload(msg);
+        return buffer != null ? new JsonObject(buffer.toString(StandardCharsets.UTF_8)) : null;
     }
 
     /**
-     * Gets a message's body as String object that can be used for constructing a JsonObject or bind a POJO using
-     * jackson-databind e.g.
+     * Gets a message's body as Buffer object.
      *
      * @param msg The AMQP 1.0 message to parse the body of.
-     * @return The message body parsed into a JSON object or {@code null} if the message does not have a <em>Data</em>
+     * @return The message body as a Buffer or {@code null} if the message does not have a <em>Data</em>
      *         nor an <em>AmqpValue</em> section.
      * @throws NullPointerException if the message is {@code null}.
      */
-    public static String getPayload(final Message msg) {
+    public static Buffer getPayload(final Message msg) {
```

Comment:
this is a breaking change of the public API and should therefore be at least documented in the release notes.